python/py-transformers/Portfile

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           python 1.0
+
+github.setup        huggingface transformers 2.1.1 v
+revision            0
+name                py-${github.project}
+categories-append   textproc
+
+license             Apache-2
+maintainers         nomaintainer
+platforms           darwin
+supported_archs     noarch
+
+description         State-of-the-art Natural Language Processing for\
+                    TensorFlow 2.0 and PyTorch
+long_description \
+    🤗 Transformers (formerly known as pytorch-transformers and\
+    pytorch-pretrained-bert) provides state-of-the-art general-purpose\
+    architectures (BERT, GPT-2, RoBERTa, XLM, DistilBert, XLNet,\
+    CTRL...) for Natural Language Understanding (NLU) and Natural\
+    Language Generation (NLG) with over 32+ pretrained models in 100+\
+    languages and deep interoperability between TensorFlow 2.0 and\
+    PyTorch.
+
+homepage            https://huggingface.co/transformers/
+
+python.versions     37 38
+
+checksums           rmd160  708151f94aa3a805e2f9c5b845d4d4ae096a5bc5 \
+                    sha256  48c919a658e6429045b678fddb40eb6401c6be66b4d3a12a7c738da5663d952b \
+                    size    808428
+
+if {${name} ne ${subport}} {
+    # see https://github.com/huggingface/transformers/blob/master/setup.py
+    depends_lib-append \
+                    port:py${python.version}-setuptools
+    depends_run-append \
+                    port:py${python.version}-boto3 \
+                    port:py${python.version}-numpy \
+                    port:py${python.version}-protobuf3 \
+                    port:py${python.version}-regex \
+                    port:py${python.version}-requests \
+                    port:py${python.version}-sacremoses \
+                    port:py${python.version}-sentencepiece \
+                    port:py${python.version}-tqdm
+
+    depends_test-append \
+                    port:py${python.version}-pytest
+
+    test.run        yes
+    test.cmd        pytest-${python.version}
```

Comment:
so, you're kind of supposed to test things before you submit a PR. If you do ```sudo port -dv test py37-transformers``` you'll see that the tests fail because ```test.cmd``` is incorrect. I have posted before in a review comment in this same PR the correct way of adding the tests, it should be: ```test.cmd        py.test-${python.branch}```
and you will need to set the environment so that it will work also when running the tests without having the port installed: ```test.env        PYTHONPATH=${worksrcpath}/build/lib```

Reply:
but even after changing that it will not run because one will need py-torch or py-tensorflow. I'm fine with leaving that as is; the user can/should just install one of these and then will be able to run the tests.

Reply:
Thanks done.

Reply:
Your ```test.cmd``` is still incorrect... I shouldn't really need to repeat this for a third time ;) Also, as I said before, you should really test these things yourself before submitting a PR and again after you changes things.

Reply:
@reneeotten If that’s not correct, please post the correct command.

I do not posses your specialized knowledge of MacPorts Python packages.

I did test it, and it runs fine locally for me.

> Your `test.cmd` is still incorrect... I shouldn't really need to repeat this for a third time ;) Also, as I said before, you should really test these things yourself before submitting a PR and again after you changes things.

Reply:
@reneeotten Also,
```bash
which py.test-3.7
echo $?
1
```

throws an error—there is no `py.test-${python.branch}` command as far as I can tell.

Reply:
@reneeotten Furthermore, from https://github.com/huggingface/transformers#tests, upstream says that the test is:
```bash
python -m pytest -sv ./transformers/tests/
python -m pytest -sv ./examples/
```

This is what I attempted to implement in the Portfile. I do not understand why you say it is incorrect.

Reply:
@essandess I am confused here. Did you try the tests with ```sudo port -dv test py37-transformers``` or did you type ```python -m pytest -sv ./transformers/tests/```? These are two different things. 

What I see here with your Portfile in the first case: ```sh: pytest-37: command not found```. Indeed, what's installed with the ```py-pytest``` ports is for example, ```py.test-3.7``` and ```pytest-3.7``` (I didn't realize we installed both commands, so you can indeed use "pytest" as you did). However, to get the "3.7" part you should use ```${python.branch}```, in your case with ```${python.version}``` you'll get "37", which cannot work.

Reply:
Thanks for the explanation. I used `port test` on a separate box, and believe I made some syncing mistake prior to the commit.